src/System.Net.Ping/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -88,15 +88,27 @@ private SocketConfig GetSocketConfig(IPAddress address, byte[] buffer, int timeo
 
         private Socket GetRawSocket(SocketConfig socketConfig)
         {
+            IPEndPoint ep = socketConfig.EndPoint as IPEndPoint;
+
             // Setting Socket.DontFragment and .Ttl is not supported on Unix, so socketConfig.Options is ignored.
-            AddressFamily addrFamily = ((IPEndPoint)socketConfig.EndPoint).Address.AddressFamily;
+            AddressFamily addrFamily = ep.Address.AddressFamily;
             Socket socket = new Socket(addrFamily, SocketType.Raw, socketConfig.ProtocolType);
             socket.ReceiveTimeout = socketConfig.Timeout;
             socket.SendTimeout = socketConfig.Timeout;
             if (socketConfig.Options != null && socketConfig.Options.Ttl > 0)
             {
                 socket.Ttl = (short)socketConfig.Options.Ttl;
             }
+
+#pragma warning disable 618
+            // Disable warning about obsolete property. We could use GetAddressBytes but that allocates.
+            if (!ep.Address.IsIPv6Multicast && !(addrFamily == AddressFamily.InterNetwork && (ep.Address.Address & 0xf0) == 0xe0))
```

Comment:
What's the purpose of the 0xe0 check here?  A comment would be helpful.

Reply:
that verifies leading 1110 bit sequence for multicast IP range. I'll add some comment to the code. I agree that it may not be clear when only reading the code. 
I wish we have IsIPv4Multicast like we have for IPv6 but I did not see anything.